test/src/example.cpp

```diff
@@ -137,6 +138,19 @@ void Example::emit_custom_signal(const String &name, int value) {
 	emit_signal("custom_signal", name, value);
 }
 
+// Nodes need to be passed as Object
+// Classes, that inherit from RefCounted, need to be passed as Ref<>
+void Example::set_texture_of_sprite(Object *p_sprite, Ref<Texture2D> p_texture) {
+	// make sure, that the object is indeed of the intended Class
+	Sprite2D *sprite = p_sprite->cast_to<Sprite2D>(p_sprite);
```

Comment:
I don't know about `godot-cpp` specifically but in the Godot codebase `Object::cast_to` is static so it's usually called like this:
```suggestion
	Sprite2D *sprite = Object::cast_to<Sprite2D>(p_sprite);
```
Which here would also prevent crashing if `p_sprite` is `nullptr`.